app.rb

```diff
@@ -0,0 +1,84 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/user'
+require './lib/post'
+require './lib/comment'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    @user = User.find_by_id(session[:user_id])
+    @post = Post.order(created_at: :desc).pluck(:peep, :created_at, :id, :user_id)
+    @comment = Comment.order(created_at: :asc)
+    erb :index
+  end
+
+  post '/add_post' do
+    new_post = Post.new(peep: params[:peep], user_id: session[:user_id])
+    new_post.save
+    redirect '/'
+  end
+
+  get '/sign-up' do
+    erb :sign_up
+  end
+
+  get '/add-comment/:id' do
+    @post_id = params[:id]
+    @post_text = Post.find_by_id(@post_id).peep
+    @post_user_id = Post.find_by_id(@post_id).user_id
+    @username = User.find_by_id(@post_user_id).username
+    erb :new_comment
+  end
+
+  post '/commenting/:id' do
+    Comment.create(comment: params[:comment], post_id: params[:id], user_id: session[:user_id])
+    redirect "/"
+  end
+
+  post '/signing-up' do
+    @new_user = User.create(
+      username: params[:username], 
+      email: params[:email],
+      first_name: params[:first_name],
+      last_name: params[:last_name],
+      password: params[:password]
+    )
+    @new_user.save
```

Comment:
The `.create` method already saves the new record into the DB, so you don't need to call `save` afterwards

Reply:
Ah, that's silly of me